src/Hydrators/ChildHydrator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Choredo\Hydrators;
+
+use Assert\Assert;
+use Assert\Assertion;
+use Choredo\Entities;
+use Choredo\Entities\Family;
+use Choredo\FamilyAware;
+use Choredo\JsonApi\JsonApiResource;
+use Ramsey\Uuid\Uuid;
+use const Choredo\HEX_COLOR_REGEX;
+use const Choredo\SHORT_DATA_FIELD_MAX_SIZE;
+
+class ChildHydrator implements Hydrator, FamilyAware
+{
+    /**
+     * @var \Choredo\Entities\Family
+     */
+    private $family;
+
+    public function hydrate(JsonApiResource $resource)
+    {
+        $this->validateResource($resource);
+
+        return new Entities\Child(
+            $resource->getId() === JsonApiResource::TYPE_NEW ? Uuid::uuid4() : Uuid::fromString($resource->getId()),
```

Comment:
Might we worth throwing this in a static helper class or summat. 

`ResourceIdGenerator::generateId($resource)`

or a global function. whatevs.